src/components/ForexRatesTab.js

```diff
@@ -68,15 +74,48 @@ function ForexRatesTab(props) {
             description: '',
             onConfirm: () => {},
           });
-        } catch (error) {
-          setConfirmDialog({
-            visible: false,
-            description: '',
-            onConfirm: () => {},
-          });
           setSnackBarParams({
-            show: true, message: 'The Forex rate could not be set', variant: 'error',
+            show: true,
+            message: 'The Forex rate was successfully set',
+            variant: 'success',
+            action: 'close',
           });
+          setForexRates([{
+            rate: stringRateFromDecimalRateAndInteger(4, rate),
+            startTime,
+            endTime,
+            reuse: true,
+          }, ...forexRates]);
+        } catch (error) {
+          if (error instanceof SyntaxError) {
+            setConfirmDialog({
+              visible: false,
+              description: '',
+              onConfirm: () => {},
+            });
```

Comment:
I notice the pattern used elsewhere is `setConfirmDialogVisible(false)`. I can see that could cause a problem later. But perhaps the general pattern is worth factoring out into something like `clearConfirmDialog` which does as these four lines do.